### ISSUE TITLE
Deal with cubed-sphere emissions in GCHP

### ISSUE THREAD

**Issue by ChiLi90** [geoschem/geos-chem#1770]
### Name and Institution (Required)

Name: Chi Li
Institution: Washington University in St. Louis

### Confirm you have reviewed the following documentation

- [Y] [Support guidelines](https://geos-chem.readthedocs.io/en/stable/reference/SUPPORT.html)
- [Y] [User manuals](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/related-docs.html)
- [Y] [Debugging GEOS-Chem and HEMCO errors](https://geos-chem.readthedocs.io/en/stable/geos-chem-shared-docs/supplemental-guides/debug-guide.html)
- [Y] [Current and past Github issues](https://github.com/geoschem/geos-chem/issues)

### Description of your issue or question

In an inverse modeling work based on GCHP, we have generated emissions that is in cubed-sphere grid (e.g., C180 with a stretch factor of 3), but we do not know how to pass the cubed-sphere data to HEMCO and drive GCHP simulations.

The most confusing part is that in each HEMCO line in the Configuration file, we need to define if the data is 2-D (xy) or 3-D (xyz). But in a cubed-sphere configuration, a "2-D" fields has an additional dimension (face) other than "x" and "y". Should we concatenate the file, or there are other standard ways to implement a cubed sphere data into HEMCO?

Thank you.
Chi

Please provide as much detail as possible. Always include the GEOS-Chem version number and any relevant configuration and log files.

**Comment by lizziel** [comment]
Hi @ChiLi90,

Are you trying to input the cubed sphere emissions data to GCHP or to HEMCO standalone? For GCHP you can specify cubed-sphere data in `ExtData.rc` and MAPL will regrid it to whatever the run grid is. The filename in `HEMCO_Config.rc` is ignored since HEMCO does not handle the reading and regridding of the data in GCHP. For HEMCO standalone you cannot input cubed-sphere data. HEMCO does not have regridding capability beyond lat-lon to lat-lon.

**Comment by ChiLi90** [comment]
> Hi Liz,

Thanks for the prompt reply. I am a little puzzled, if I only indicate the cubed-sphere file into ExtData.rc without registering it in HEMCO_Config.rc, how can GCHP know that this file is feeding, e.g., (anthropogenic/volcanic) NOx/SO2 emissions? to the model.

If I still need to register it in HEMCO_Config.rc, should I just indicate it as "xy" even if the "face" dimension presents?

Thank you,
Chi



> Hi @ChiLi90,
> 
> Are you trying to input the cubed sphere emissions data to GCHP or to HEMCO standalone? For GCHP you can specify cubed-sphere data in `ExtData.rc` and MAPL will regrid it to whatever the run grid is. The filename in `HEMCO_Config.rc` is ignored since HEMCO does not handle the reading and regridding of the data in GCHP. For HEMCO standalone you cannot input cubed-sphere data. HEMCO does not have regridding capability beyond lat-lon to lat-lon.

**Comment by yantosca** [comment]
Hi @ChiLi90: In ExtData.rc we define an export name (the first column), plus the various file settings, like units, how often to read it, which variable to read, filename, etc.  The file is read and regridded by MAPL.
```console
#--------|-------|------|------------|---------------|--------|-------|---------------------|
# Export |       |      |            |___ Refresh ___|____ Factors ___|___ External File ___|
# Name   | Units | Clim |Conservative| Time Template | Offset | Scale | Variable | Template |
#--------|-------|------|------------|---------------|--------|-------|----------|----------|

CEDS_NO_AGR  kg/m2/s N Y F%y4-%m2-01T00:00:00 none none NO_agr  ./HcoDir/CEDS/v2021-06/%y4/NO-em-anthro_CMIP_CEDS_%y4.nc
```
Then GCHP will match the export name to the container name in HEMCO_Config.rc:
```console
0 CEDS_NO_AGR     $ROOT/CEDS/v2021-06/$YYYY/NO-em-anthro_CMIP_CEDS_$YYYY.nc            NO_agr            1750-2019/1-12/1/0 C xy kg/m2/s NO    25        1 5
```
So as long as there is an entry for e.g.`CEDS_NO_AGR` in both HEMCO_Config.rc and ExtData,rc, then the GCHP interface to HEMCO will take the file pointer to this data from MAPL and pass it to HEMCO.  HEMCO will put it into the NO species and then apply the scale factors, category, hierarchy, and masks (if any).

So basically GCHP ignores the file I/O stuff that is specified in HEMCO_Config.rc, because that part is done by MAPL.  We typically keep the HEMCO_Config.rc files in both GCClassic and GCHP as similar as possible so that they are easier to compare.

In the future we would like to abstract the file I/O information out of HEMCO_Config.rc into a YAML file that could be read both by GCHP and GCClassic.  We are a ways off from that happening, though.

**Comment by lizziel** [comment]
See also the entry about `HEMCO_Config.rc` in GCHP ReadTheDocs: https://gchp.readthedocs.io/en/latest/user-guide/config-files/HEMCO_Config_rc.html

**Comment by ChiLi90** [comment]
> https://gchp.readthedocs.io/en/latest/user-guide/config-files/HEMCO_Config_rc.html

Thank you very much @lizziel and @yantosca. We will try to implement it following these useful guides.

Chi

**Comment by lizziel** [comment]
I am closing this issue now. If you still have problems please open a new issue.
